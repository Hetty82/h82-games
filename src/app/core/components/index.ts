import { AppComponent } from './app/app.component'
import { LayoutComponent } from './layout/layout.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const components: any[] = [
  AppComponent,
  LayoutComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './layout/layout.component'
export * from './not-found/not-found.component'
