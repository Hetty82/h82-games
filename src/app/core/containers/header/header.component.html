<header class="header">
  <div class="left">
    <h1 class="title">{{ title }}</h1>
    <p>{{ subTitle }}</p>
  </div>

  <div class="center">
    <a routerLink="/" routerLinkActive="active">Home</a>
  </div>

  <div class="right">
    <ng-container *ngIf="(user$ | async) as user; else login">
      {{ user.name }}
      <button (click)="logout()" class="button -warning">Logout</button>
    </ng-container>

    <ng-template #login>
      <a routerLink="/login" routerLinkActive="active">Login</a>
    </ng-template>
  </div>
</header>
