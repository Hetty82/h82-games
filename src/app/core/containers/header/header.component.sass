@import '~styles/variables'

.header
  display: flex
  align-items: center
  margin-bottom: $spacing-xxl
  padding: $spacing-sm $spacing-default
  border-bottom: $border-light-double

.center
  flex-basis: 60%
  text-align: center

.left,
.right
  flex-basis: 20%

.right
  text-align: right

.title
  font-size: $font-size-lg
  margin-bottom: 0
