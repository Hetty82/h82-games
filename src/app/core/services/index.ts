import { GamesService } from './games.service'
import { UserService } from './user.service'

export const services: any[] = [
  GamesService,
  UserService,
]

export * from './games.service'
export * from './user.service'
