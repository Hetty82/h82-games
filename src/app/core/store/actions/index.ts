export * from './games.actions'
export * from './user.actions'
