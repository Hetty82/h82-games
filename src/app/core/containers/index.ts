import { GamesComponent } from './games/games.component'
import { HeaderComponent } from './header/header.component'
import { LoginComponent } from './login/login.component'

export const containers: any[] = [
  GamesComponent,
  HeaderComponent,
  LoginComponent,
]

export * from './games/games.component'
export * from './header/header.component'
export * from './login/login.component'
