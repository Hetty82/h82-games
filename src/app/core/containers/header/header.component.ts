import { Component, OnInit, Input } from '@angular/core'
import { Store, select } from '@ngrx/store'

import * as fromRoot from '../../../store'


@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.sass'],
})
export class HeaderComponent implements OnInit {
  @Input() title: string
  @Input() subTitle: string

  user$ = this.store.pipe(select(fromRoot.getCurrentUser))

  constructor(private store: Store<fromRoot.State>) {
  }

  ngOnInit() {
  }

  logout() {
    this.store.dispatch(new fromRoot.DeselectUser())
  }
}
