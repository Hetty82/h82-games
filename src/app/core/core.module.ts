import { NgModule } from '@angular/core'
import { CommonModule } from '@angular/common'
import { RouterModule } from '@angular/router'

import * as fromComponents from './components'
import * as fromContainers from './containers'
import * as fromServices from './services'


@NgModule({
  imports: [
    CommonModule,
    RouterModule,
  ],
  declarations: [
    ...fromComponents.components,
    ...fromContainers.containers,
  ],
  exports: [
    fromContainers.AppComponent,
  ],
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: fromServices.services,
    }
  }
}
