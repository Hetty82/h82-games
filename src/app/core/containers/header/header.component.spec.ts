import { NO_ERRORS_SCHEMA } from '@angular/core'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Store } from '@ngrx/store'

import { HeaderComponent } from './header.component'


describe('HeaderComponent', () => {
  let component: HeaderComponent
  let fixture: ComponentFixture<HeaderComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ HeaderComponent ],
      providers: [
        {
          provide: Store,
          useValue: {
            pipe: jest.fn(),
          },
        }
      ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
