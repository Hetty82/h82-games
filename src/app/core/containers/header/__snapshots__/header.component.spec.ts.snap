// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot 1`] = `
<app-header
  store={[Function Object]}
  user$="undefined"
>
  <header
    class="header"
  >
    <div
      class="left"
    >
      <h1
        class="title"
      >
        
      </h1>
      <p>
        
      </p>
    </div>
    <div
      class="center"
    >
      <a
        routerlink="/games"
        routerlinkactive="active"
      >
        Games
      </a>
    </div>
    <div
      class="right"
    >
      
      <a
        routerlink="/login"
        routerlinkactive="active"
      >
        Login
      </a>
      
    </div>
  </header>
</app-header>
`;
